File: src/main/java/com/fouan/actors/DangerLevel.java
package com.fouan.actors;

public enum DangerLevel {
    BLUE(6),
    YELLOW(18),
    ORANGE(42),
    RED(43);

    private final int experienceThreshold;

    DangerLevel(int experienceThreshold) {
        this.experienceThreshold = experienceThreshold;
    }

    public static DangerLevel fromExperience(int experience) {
        for (var value : values()) {
            if (experience <= value.experienceThreshold) {
                return value;
            }
        }

        throw new IllegalStateException("No danger level found");
    }
}

File: src/main/java/com/fouan/actors/DangerLevel.kt
package com.fouan.actors

enum class DangerLevel(private val experienceThreshold: Int) {
    BLUE(6),
    YELLOW(18),
    ORANGE(42),
    RED(43);

    companion object {
        @JvmStatic
        fun fromExperience(experience: Int): DangerLevel {
            for (value in values()) {
                if (experience <= value.experienceThreshold) {
                    return value
                }
            }
            throw IllegalStateException("No danger level found")
        }
    }
}
